servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.kt

```diff
@@ -60,45 +61,43 @@ const val DEFAULT_INSTRUMENTATION_ORDER = 0
     FederatedSchemaAutoConfiguration::class
 )
 class GraphQLSchemaConfiguration {
-
     @Bean
     @ConditionalOnMissingBean
     fun graphQL(
         schema: GraphQLSchema,
         dataFetcherExceptionHandler: DataFetcherExceptionHandler,
-        instrumentations: Optional<List<Instrumentation>>,
+        providedInstrumentations: Optional<List<Instrumentation>>,
         executionIdProvider: Optional<ExecutionIdProvider>,
         preparsedDocumentProvider: Optional<PreparsedDocumentProvider>,
         config: GraphQLConfigurationProperties,
         idValueUnboxer: IDValueUnboxer
     ): GraphQL {
-        val graphQL = GraphQL.newGraphQL(schema)
+        val graphQLBuilder = GraphQL.newGraphQL(schema)
             .queryExecutionStrategy(AsyncExecutionStrategy(dataFetcherExceptionHandler))
             .mutationExecutionStrategy(AsyncSerialExecutionStrategy(dataFetcherExceptionHandler))
             .subscriptionExecutionStrategy(FlowSubscriptionExecutionStrategy(dataFetcherExceptionHandler))
             .valueUnboxer(idValueUnboxer)
-
-        instrumentations.ifPresent { unordered ->
-            if (unordered.size == 1) {
-                graphQL.instrumentation(unordered.first())
-            } else {
-                val sorted = unordered.sortedBy {
-                    if (it is Ordered) {
-                        it.order
-                    } else {
-                        DEFAULT_INSTRUMENTATION_ORDER
-                    }
-                }
-                graphQL.instrumentation(ChainedInstrumentation(sorted))
+            .also { builder ->
+                executionIdProvider.ifPresent(builder::executionIdProvider)
+                preparsedDocumentProvider.ifPresent(builder::preparsedDocumentProvider)
             }
+
+        val instrumentations = mutableListOf<Instrumentation>()
```

Comment:
if we already introducing that `also` block above -> should we move this inside as well?